syntax/abstract.mjs

```diff
@@ -43,6 +43,15 @@ export function list_into(Type) {
                         .filter(remove_blank_lines)));
         case FTL.SelectExpression:
             return ([selector, variants]) => {
+                let invalid_selector_found =
+                    selector instanceof FTL.MessageReference
+                    || selector instanceof FTL.TermReference
+                    || selector instanceof FTL.VariantExpression
+                    || (selector instanceof FTL.AttributeExpression
+                        && selector.ref instanceof FTL.MessageReference);
```

Comment:
I found this hard to digest. I wonder if a positive criteria is revealing the intent here better? I.e, instead of excluding `MessageReference`, include `StringLiteral`? That would also allow to return early.

It's yet one more condition, though. Also a reason why I'm still torn on the outside-impact on this change.

Reply:
I'd prefer to keep it as is for consistency with other rules. The rules highlight the invalid cases which are generally less frequent.